fluss-lake/fluss-lake-format-paimon/src/main/java/com/alibaba/fluss/lake/paimon/PaimonLakeCatalog.java

```diff
@@ -0,0 +1,158 @@
+            }
+        }
+    }
+
+    private void createTable(Identifier tablePath, Schema schema)
+            throws Catalog.DatabaseNotExistException {
+        try {
+            // not ignore if table exists
+            paimonCatalog.createTable(tablePath, schema, false);
+        } catch (Catalog.TableAlreadyExistException e) {
+            throw new TableAlreadyExistException("Table " + tablePath + " already exists.");
+        }
+    }
+
+    private void createDatabase(String databaseName) {
+        try {
+            // ignore if exists
+            paimonCatalog.createDatabase(databaseName, true);
+        } catch (Catalog.DatabaseAlreadyExistException e) {
+            // do nothing, shouldn't throw since ignoreIfExists
+        }
+    }
+
+    private Identifier toPaimonIdentifier(TablePath tablePath) {
+        return Identifier.create(tablePath.getDatabaseName(), tablePath.getTableName());
+    }
+
+    private Schema toPaimonSchema(TableDescriptor tableDescriptor) {
+        Schema.Builder schemaBuilder = Schema.newBuilder();
+        Options options = new Options();
+
+        // When bucket key is undefined, it should use dynamic bucket (bucket = -1) mode.
+        List<String> bucketKeys = tableDescriptor.getBucketKeys();
+        if (!bucketKeys.isEmpty()) {
+            int numBuckets =
+                    tableDescriptor
+                            .getTableDistribution()
+                            .flatMap(TableDescriptor.TableDistribution::getBucketCount)
+                            .orElseThrow(
+                                    () ->
+                                            new IllegalArgumentException(
+                                                    "Bucket count should be set."));
+            options.set(CoreOptions.BUCKET, numBuckets);
+            options.set(CoreOptions.BUCKET_KEY, String.join(",", bucketKeys));
+        } else {
+            options.set(CoreOptions.BUCKET, CoreOptions.BUCKET.defaultValue());
+        }
+
+        // set schema
+        for (com.alibaba.fluss.metadata.Schema.Column column :
+                tableDescriptor.getSchema().getColumns()) {
+            schemaBuilder.column(
+                    column.getName(),
+                    column.getDataType().accept(FlussDataTypeToPaimonDataType.INSTANCE),
+                    column.getComment().orElse(null));
+        }
+
+        // set pk
+        if (tableDescriptor.getSchema().getPrimaryKey().isPresent()) {
+            schemaBuilder.primaryKey(
+                    tableDescriptor.getSchema().getPrimaryKey().get().getColumnNames());
+            options.set(
+                    CoreOptions.CHANGELOG_PRODUCER.key(),
+                    CoreOptions.ChangelogProducer.INPUT.toString());
+        } else {
+            // for log table, need to set bucket, offset and timestamp
+            schemaBuilder.column(BUCKET_COLUMN_NAME, DataTypes.INT());
```

Comment:
I mean we need to  check the original schema contains same system column name like `__bucket` or not, to avoid conflict with users' original columns.